tests/io/test_OutputFile.py

```diff
@@ -0,0 +1,694 @@
+import sys
+from contextlib import contextmanager
+from io import StringIO
+from typing import Optional
+
+import netCDF4
+import numpy
+import pytest
+from mock import MagicMock, patch
+
+from dfastmi.io.OutputFile import OutputFile
+
+
+class TestOutputFile(OutputFile):
+    def x_velocity(
+        self,
+        time_index_from_last: Optional[int] = None,
+    ) -> numpy.ndarray:
+        return numpy.array([[0.0], [1.0], [2.0]])
+
+    def y_velocity(
+        self,
+        time_index_from_last: Optional[int] = None,
+    ) -> numpy.ndarray:
+        return numpy.array([[0.0], [1.0], [2.0]])
+
+    def water_depth(
+        self,
+        time_index_from_last: Optional[int] = None,
+    ) -> numpy.ndarray:
+        return numpy.array([[0.0], [1.0], [2.0]])
+
+
+def open_test_output_file() -> TestOutputFile:
+    filename = "tests/files/"
+    return TestOutputFile(filename)
+
+
+@pytest.fixture
+def test_output_file() -> OutputFile:
+    return open_test_output_file()
+
+
+@contextmanager
+def captured_output():
+    new_out, new_err = StringIO(), StringIO()
+    old_out, old_err = sys.stdout, sys.stderr
+    try:
+        sys.stdout, sys.stderr = new_out, new_err
+        yield sys.stdout, sys.stderr
+    finally:
+        sys.stdout, sys.stderr = old_out, old_err
+
+
+class Test_OutputFile:
+    @pytest.mark.parametrize(
+        "data_file, dataref",
+        [
+            (open_test_output_file(), 1.0),
+        ],
+    )
+    def test_x_velocity(self, data_file: test_output_file, dataref: float):
```

Comment:
is it needed to test this (x_velocity, y_velocity and water_depth), since this is an abstract method, now you are just testing your setup class in the test.

Reply:
discussed, this is not data access testing but mocking / stubbing of abstract object. It shows developer more quickly how the methods are supposed to be used instead of trying to read the data from the netcdf file